python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,240 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import shutil
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from tempfile import TemporaryDirectory
+from typing import Dict, List
+
+from revng.model import Binary, StructBase, get_element_by_path  # type: ignore[attr-defined]
+from revng.tupletree import TypedList
+
+
+class Project(ABC):
+    """
+    This class is the basis for building a project. It defines
+    abstract methods that the inherited classes have to implement
+    and it also implements some common methods.
+    """
+
+    def __init__(self):
+        self.model: Binary = Binary()
+        self.last_saved_model: Binary = Binary()
+        self.pipeline_description: Dict = {}
+        self.input_binary: str
+        self.revng_executable: str
+        self.resume_dir: str
```

Comment:
No abbreviations, `resume_path` or `resume_directory_path`.

Reply:
Renamed to `resume_path`